core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -108,6 +110,19 @@ public CommitFailedException wrapped() {
     }
   }
 
+  private static <T> Pair<List<T>, String> paginate(List<T> list, String pageToken, int pageSize) {
```

Comment:
For pageToken, it seems we can use integer instead of String so that there is no need to convert between integer and String.

Reply:
The public API as described in the OpenAPI spec defines a string: https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml

A string is generally more preferable as it allows the implementation more freedom to use different paging techniques than just row number based (for example cursor based).

Reply:
thanks for the clarification.